azure-proto-core/Resources/Plan.cs

```diff
@@ -27,82 +27,17 @@ public int CompareTo(Plan other)
                 return 1;
             }
 
-            if (Name == null && other.Name != null)
+            int compareResult = 0;
+            if ((compareResult = string.Compare(Name, other.Name)) == 0 &&
```

Comment:
Please add StringComparison.InvariantCultureIgnoreCase as 3rd param to all string.Compare calls.